code/_onclick/hud/screen_objects.dm

```diff
@@ -84,18 +147,86 @@
 	return 1
 
 
+/obj/screen/stamina
+	name = "stamina"
+	icon = 'icons/effects/progessbar.dmi'
+	icon_state = "prog_bar_100"
+	invisibility = INVISIBILITY_MAXIMUM
+	screen_loc = ui_stamina
+
+/obj/screen/inventory/hand
+	name = "l_hand"
+	icon_state = "hand_l"
+	screen_loc = ui_lhand
+	var/hand_tag = "l"
+
+/obj/screen/inventory/hand/update_icon(active = FALSE)
+	cut_overlays()
+	if(active)
+		add_overlay("hand_active")
+
+/obj/screen/inventory/hand/Click()
+	if(world.time <= usr.next_move)
+		return TRUE
+	if(usr.incapacitated() || !iscarbon(usr))
+		return TRUE
+	var/mob/living/carbon/C = usr
+	C.activate_hand(hand_tag)
+
+/obj/screen/inventory/hand/right
+	name = "r_hand"
+	icon_state = "hand_r"
+	screen_loc = ui_rhand
+	hand_tag = "r"
+
+/obj/screen/swap_hand
+	name = "swap hand"
+	name = "swap"
+	icon_state = "hand1"
+	screen_loc = ui_swaphand1
+
+/obj/screen/swap_hand/Click()
+	if(!iscarbon(usr))
+		return
+	var/mob/living/carbon/M = usr
+	M.swap_hand()
+
+/obj/screen/swap_hand/right
+	icon_state = "hand2"
+	screen_loc = ui_swaphand2
+
+/obj/screen/swap_hand/human
+	icon_state = "hand1"
 
+/obj/screen/toggle_inv
+	name = "toggle"
+	icon = 'icons/hud/midnight.dmi'
+	icon_state = "toggle"
+	screen_loc = ui_inventory
 
+/obj/screen/toggle_inv/Click()
+	if(usr.hud_used.inventory_shown)
+		usr.hud_used.inventory_shown = FALSE
+		usr.client.screen -= usr.hud_used.toggleable_inventory
+	else
+		usr.hud_used.inventory_shown = TRUE
+		usr.client.screen += usr.hud_used.toggleable_inventory
 
+	usr.hud_used.hidden_inventory_update()
 
 /obj/screen/intent
 	name = "intent"
-	icon = 'icons/mob/screen1_White.dmi'
+	icon = 'icons/hud/White.dmi'
 	icon_state = "intent_help"
 	screen_loc = ui_acti
 	var/intent = I_HELP
 
-/obj/screen/intent/Click(var/location, var/control, var/params)
+/obj/screen/intent/New(mob/C)
+	if (C)
```

Comment:
doing this in added_to_screen worked fine, putting code in new isnt a great idea. Especially when you didn't call parent!